pkg/operator/clustermemberremovalcontroller/clustermemberremovalcontroller.go

```diff
@@ -177,19 +177,50 @@ func (c *clusterMemberRemovalController) attemptToScaleDown(ctx context.Context,
 	if err != nil {
 		return fmt.Errorf("could not get a list of unhealthy members: %v", err)
 	}
+
+	// map machines pending deletion to node internal ip
+	votingMachinesPendingDeletionIndex := ceohelpers.IndexMachinesByNodeInternalIP(votingMachinesPendingDeletion)
```

Comment:
here the usage seems safe, but please add a unit test to exercise a node with multiple IPs pending deletion. 

also cc @geliu2016 fyi for verifying on dualstack vsphere

Reply:
>  can you ensure this PR also works on dualstack setups?
 sure, I am trying to create a cluster with this PR atop, but I need to use the installer, once I test this i will try with dual stack 👍🏽

Reply:
> but please add a unit test to exercise a node with multiple IPs pending deletion. 
I am working on the test now, will add this 

Do you know if there is a similar test using dual stack somewhere ? just I wanna see how to create the scenario to avoid confusion :)

Reply:
I added one in the above PR:
https://github.com/openshift/cluster-etcd-operator/pull/858/files#diff-b3ed5284128aa9fb29b8b026649f6401ca9e67922974c15ddff2a795d0225d76R345

Reply:
> here the usage seems safe, but please add a unit test to exercise a node with multiple IPs pending deletion.
> 
> also cc @geliu2016 fyi for verifying on dualstack (eg vsphere or so)
@tjungblu I tried dualstack on baremetal, and destroy one etcd-pod.yaml to make etcd member in unhealth status, then try to create a new machine to enable N+1 master node, but unhealth etcd member failed to be deleted automatically.

# oc get machine
NAME                              PHASE         TYPE   REGION   ZONE   AGE
geliudual2-5dss6-master-0         Failed                               3h23m
geliudual2-5dss6-master-03        Provisioned                          32m
geliudual2-5dss6-master-1         Running                              3h23m
geliudual2-5dss6-master-2         Running                              3h23m
geliudual2-5dss6-worker-0-5kvmd   Running                              177m
geliudual2-5dss6-worker-0-fv99s   Running                              177m

# oc get machine geliudual2-5dss6-master-0 -o yaml
.................
status:
  conditions:
  - lastTransitionTime: "2022-10-10T05:14:40Z"
    message: 'Drain operation currently blocked by: [{Name:EtcdQuorumOperator Owner:clusteroperator/etcd}]'
    reason: HookPresent
    severity: Warning
    status: "False"
    type: Drainable
  - lastTransitionTime: "2022-10-10T07:37:13Z"
    message: Instance not found on provider
    reason: InstanceMissing
    severity: Warning
    status: "False"
    type: InstanceExists
  - lastTransitionTime: "2022-10-10T05:12:03Z"
    status: "True"
    type: Terminable
  errorMessage: Can't find created instance.

Reply:
@geliu2016 Thanks for your follow up 

So this PR tackles only the case when we have N+1 master machines, and then one goes unhealthy .. 

The scenario you have tested is another [issue](https://issues.redhat.com/browse/ETCD-328), i am still working on it. However, the result of your test is as expected